kube-client/src/config/file_config.rs

```diff
@@ -497,6 +497,33 @@ where
     });
 }
 
+fn read_path<P: AsRef<Path>>(path: P) -> io::Result<String> {
+    let bytes = fs::read(&path)?;
+    match bytes.as_slice() {
+        [0xFF, 0xFE, ..] => {
```

Comment:
a comment for  these magic numbers would be helpful for a future person 😅

Reply:
would you prefer using `encoding_rs` crate instead?

Reply:
Perhaps also putting a comment to use [String::from_utf16le](https://doc.rust-lang.org/stable/std/string/struct.String.html#method.from_utf16le) and friends when they become stable